app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md

```diff
@@ -0,0 +1,201 @@
+Similarly, a method named `ordinal()` returns the position of the enum constant in the declaration.
+```java
+System.out.println(DayOfWeek.MONDAY.name());    // prints "MONDAY"
+System.out.println(DayOfWeek.MONDAY.ordinal()); // prints "0" because MONDAY is the first constant in the DayOfWeek enum
+```
+
+Aside from instance methods, there are also static methods added to all enums.
+The method `values()` returns an array containing all instances of the enum and the method `valueOf(String)` can be used to get a specific instance by its name.
+```java
+DayOfWeek[] days = DayOfWeek.values(); // all days of the week
+DayOfWeek monday = DayOfWeek.valueOf("MONDAY");
+```
+
+Furthermore, enums implement the interface [`Comparable`](javadoc:Comparable).
+By default, enums are ordered according to their ordinal number
+i.e. in the order of occurrence of the enum constant.
+This allows for comparing instances of enums as well as sorting or searching.
+
+```java
+public void compareDayOfWeek(DayOfWeek dayOfWeek){
+    int comparison = dayOfWeek.compareTo(DayOfWeek.WEDNESDAY);
+    if ( comparison < 0) {
+        System.out.println("It's before the middle of the work week.");
+    } else if(comparison > 0){
+        System.out.println("It's after the middle of the work week.");
+    } else {
+        System.out.println("It's the middle of the work week.");
+    }
+}
+```
+
+```java
+List<DayOfWeek> days = new ArrayList<>(List.of(DayOfWeek.FRIDAY, DayOfWeek.TUESDAY, DayOfWeek.SATURDAY));
+Collections.sort(days);
+```
+
+
+<a id="singletons">&nbsp;</a>
+## Using enums as singletons
+
+Since enums can only have a specific number of instances, it is possible to create a singleton by creating an enum with only a single enum constant.
+```java
+public enum SomeSingleton {
+    INSTANCE;
+    //fields, methods, etc.
+}
+```
+
+<a id="abstract">&nbsp;</a>
+## Abstract methods in enums
+
+Even though enums cannot be extended, they can still have `abstract` methods. In that case, an implementation must be present in each enum constant.
+```java
+enum MyEnum {
+    A() {
+        @Override
+        void doSomething() {
+            System.out.println("a");
+        }
+    },
+    B() {
+        @Override
+        void doSomething() {
+            System.out.println("b");
+        }
+    };
+    
+    abstract void doSomething();
+}
+```
+
+<a id="conclusion">&nbsp;</a>
+## Conclusion
+
+All in all, enums provide a simple and safe way of limiting the instances of a type
+while preserving most of the flexibilities of classes.
+
+However, care should be taken when using enums where the number (or names) of instances is subject to change.
+Whenever enum constants are changed, there may be compilation errors, runtime errors or other inconsistencies
+due to other code expecting the original version of the enum.
+This is especially important in cases where the enum is also used by other people's code.
+
+Furthermore, it might be worth considering to use other options
```

Comment:
This phrase is enigmatic: there are other options when having a big number of constants in the same file. Some people would wonder how many constants is too many for an enum? and what are the other options?